internal/entitlement/client.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 Stefan Prodan.
+// SPDX-License-Identifier: AGPL-3.0
+
+package entitlement
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+)
+
+const (
+	// VendorKey is the key in the entitlement secret
+	// that holds the vendor name.
+	VendorKey = "vendor"
+
+	// TokenKey is the key in the entitlement secret
+	// that holds the token.
+	TokenKey = "token"
+
+	// DefaultVendor is the default vendor name.
+	DefaultVendor = "controlplane"
+
+	// MarketplaceTypeEnvKey is the environment variable key
+	// that holds the marketplace type.
+	MarketplaceTypeEnvKey = "MARKETPLACE_TYPE"
```

Comment:
I did not find this env var in the flags, how is it setup?

Reply:
The Helm chart used in the Marketplace sets the evn var:
- https://github.com/controlplaneio-fluxcd/charts/blob/32c902d67dd251ab1a18118b7201e420ea3c9754/charts/flux-operator/values.yaml#L86-L90
- https://github.com/controlplaneio-fluxcd/charts/blob/32c902d67dd251ab1a18118b7201e420ea3c9754/charts/flux-operator/templates/deployment.yaml#L53-L64